app/src/main/java/com/kabouzeid/gramophone/util/PreferenceUtil.java

```diff
@@ -266,11 +267,19 @@ public final String getGenreSortOrder() {
         return mPreferences.getString(GENRE_SORT_ORDER, SortOrder.GenreSortOrder.GENRE_A_Z);
     }
 
-    public long getLastAddedCutoff() {
+    public long getLastAddedCutoffTimeSecs() {
+        return getCutoffTimeMillis(LAST_ADDED_CUTOFF) / 1000;
+    }
+
+    public long getRecentlyPlayedCutoffTimeMillis() {
```

Comment:
Why is one of them in seconds and the other in milliseconds?

Reply:
It s due to the unit of time in MediaStore (seconds) vs in the internal database (millis). Best way would be to have an uniform type, but this would force creating/altering the internal db.

I ll leave a comment on the getCutoffTime explaining this discrep.

Reply:
See efd340c